### PR TITLE
Tracks to vtk on object

### PR DESCRIPTION
**Pull request by shimwell**
Currently there is a [script](https://github.com/openmc-dev/openmc/blob/develop/scripts/openmc-track-to-vtk) that one can run from the terminal on a tracks.h5 file.

This PR adds the ability to write vtk files of tracks from within the OpenMC Python API. This allows users to create the vtk file in the following manner

```python
import openmc
tracks = openmc.Tracks('tracks.h5')
tracks.write_tracks_to_vtk('tracks.vtk')
```

Part of the motivation for this is to make the production of track vtk files a little easier from jupyter notebooks with out the need for ```!``` to run scripts. ps I am also [viewing](https://itkwidgets.readthedocs.io/en/latest/integrations.html#vtk) the vtk files in jupyter so it can make a nice workshop example

### CODE REVIEW

File: openmc/tracks.py [hunk 1]
```diff
@@ -266,6 +266,55 @@ def plot(self):
             track.plot(ax)
         return ax
 
+    def write_tracks_to_vtk(self, filename):
+        """Creates a VTK file of the tracks
+
+        Parameters
+        ----------
+        filename : str
```
Comment (paulromano) [inline openmc/tracks.py:274]: support path-like too? I'm not sure vtk supports it so you might have to write `str(filename)`

File: openmc/tracks.py [hunk 2]
```diff
@@ -266,6 +266,55 @@ def plot(self):
...
+            Name of the VTK file to write.
+
+        Returns
+        -------
+        vtk.vtkStructuredGrid
+            the VTK object
```
Comment (paulromano) [inline openmc/tracks.py:280]: This needs to be updated -- I am assuming we'd want to return the `vtkPolyData` object? Also, right now, the method is returning the filename which should also be changed.

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for the PR @shimwell! With this method on the `Tracks` object, we should just use it in `openmc-tracks-to-vtk`, which has the nearly the same code. Right now, the only difference is that `openmc-tracks-to-vtk` works with multiple track files, but we already have the ability to combine multiple track files with `Tracks.combine(...)`.

File: openmc/tracks.py (hunk 1 above)
Comment (shimwell) [reply openmc/tracks.py]: Thanks paul I tried it and you were right SetFileName accepts str types only

File: openmc/tracks.py (hunk 2 above)
Comment (shimwell) [reply openmc/tracks.py]: Yep good idea, I've updated the doc string and am now returning the vtkpolydata

**Comment by shimwell** [comment]
The openmc-tracks-to-vtk script has been updated to make use of this write_tracks_to_vtk method as requested

File: tests/unit_tests/test_tracks.py [hunk 3]
```diff
@@ -141,3 +141,18 @@ def test_filter(sphere_model, run_in_tmpdir):
     assert matches == tracks
     matches = tracks.filter(particle='bunnytron')
     assert matches == []
+
+
+def test_write_tracks_to_vtk(sphere_model):
+    # Set maximum number of tracks per process to write
```
Comment (paulromano) [inline tests/unit_tests/test_tracks.py:147]:
```suggestion
def test_write_tracks_to_vtk(sphere_model):
    vtk = pytest.importorskip('vtk')
    # Set maximum number of tracks per process to write
```

File: openmc/tracks.py [hunk 4]
```diff
@@ -266,6 +266,57 @@ def plot(self):
             track.plot(ax)
         return ax
 
+    def write_tracks_to_vtk(self, filename=Path('tracks.vtk')):
```
Comment (paulromano) [inline openmc/tracks.py:269]:
```suggestion
    def write_tracks_to_vtk(self, filename=Path('tracks.vtp')):
```

File: tests/unit_tests/test_tracks.py [hunk 5]
```diff
@@ -141,3 +141,18 @@ def test_filter(sphere_model, run_in_tmpdir):
...
+    sphere_model.settings.photon_transport = True
+
+    # Run OpenMC to generate tracks.h5 file
+    generate_track_file(sphere_model, tracks=True)
+
+    tracks = openmc.Tracks('tracks.h5')
+    filename = tracks.write_tracks_to_vtk('tracks.vtk')
+
+    assert filename == 'tracks.vtk'
+    assert Path('tracks.vtk').is_file()
```
Comment (paulromano) [inline tests/unit_tests/test_tracks.py:158]:
```suggestion
    polydata = tracks.write_tracks_to_vtk('tracks.vtp')

    assert isinstance(polydata, vtk.vtkPolyData)
    assert Path('tracks.vtp').is_file()
```

File: openmc/tracks.py [hunk 6]
```diff
@@ -6,7 +6,7 @@
 from .checkvalue import check_filetype_version
 from .source import SourceParticle, ParticleType
 
-
+from pathlib import Path
```
Comment (paulromano) [inline openmc/tracks.py:9]:
```suggestion
from pathlib import Path

```

**Review by paulromano — CHANGES_REQUESTED** [review]

**Review by paulromano — APPROVED** [review]

**Pull request merged**